### PR TITLE
Permit thread state SUSPENDED to go to STOPPED

### PR DESCRIPTION
**Pull request by benoittgt**
TLDR; Otherwise you can have exception like :

```
Assertion Failed: vernier.cc:857:set_state:state == State::RUNNING || state == State::STARTED
```
---

I wanted to plug Vernier to the [benchmark](https://github.com/speedshop/threadbench) mentioned here https://github.com/rails/rails/issues/50450. While trying to plug it, I was stop by an exception from Vernier. I modified the code to permit this handling of state and I was able to [get a result](https://share.firefox.dev/3vSXggu).

It is maybe a mistake from when I tried to setup Vernier on this project. It [can be seen here](https://github.com/speedshop/threadbench/compare/main...benoittgt:threadbench:3.3?expand=1#diff-5fe74a7aca8668f86d1251d17e23b3ac9c55504d8d78698a8168a9cae87b350c).

I am not sure we want to permit this, or instead raise and error for invalid code ?

Full stacktrace can be see here: https://github.com/speedshop/threadbench/commit/7e992f2774ab197640c02a14405eee635939a45f

### CODE REVIEW

**Comment by casperisfine** [comment]
`STOPPED` in Vernier corresponds to `EXITED`?

If so, then 👍 as it's part of the spec that `SUSPENDED -> EXITED` is possible: https://github.com/ruby/ruby/blob/8bff7e996cf65159b4ed7b55c284de6651b7e637/test/-ext-/thread/helper.rb#L42-L45

**Comment by benoittgt** [comment]
Thanks for pointing this code. I was looking for reference to what is "STOPPED" in Ruby codebase.

**Comment by jhawthorn** [comment]
Thank you! I'll try to cut a release soon

Do you happen to know the code which causes this pattern. It would be great to have a test case for it.

**Comment by benoittgt** [comment]
@jhawthorn not yet but I will be interested to dig in. Because it was during a Puma benchmark I am curious to see when it was.
